### PR TITLE
Passphrase command line parameter

### PR DESCRIPTION
**Pull request by ni4**
This PR adds --passphrase command line parameter handling, and introduces 'string' password provider. Also it updates performance test with some fixes and improvements.
It was added for easier handling of passphrase in performance and CLI tests, however most likely users will make use of it as well. Despite the fact of quite low security properties of such approach.

### CODE REVIEW

**Comment by ronaldtse** [comment]
@ni4 Maybe the problem with the performance discrepancy is that RNP is run first, so it warms the file cache for the later run of GnuPG. We could solve it by first warming the cache manually by reading it and then execute RNP/GnuPG?

File: src/tests/cli_perf.py [hunk 1]
```diff
@@ -54,7 +55,7 @@ def setup():
     ret, out, err = run_proc(GPG, ['--batch', '--passphrase', '', '--homedir', GPGDIR, '--import', path.join(RNPDIR, 'pubring.gpg'), path.join(RNPDIR, 'secring.gpg')])
 
     # Generating small file for tests
-    SMALLSIZE = 3312;
+    SMALLSIZE = 33120;
```
Comment (flowher) [inline src/tests/cli_perf.py]: is it intended?

Comment (ni4) [reply src/tests/cli_perf.py]: Nope, changed during some testing. Thanks for noticing.

**Comment by ni4** [comment]
@ronaldtse I tried to change their order and that didn't make any impact on comparison. Even if cache has some impact then it may happen only on first iteration, while on small files tests run 100 times.

**Comment by ni4** [comment]
Got 2 approvals, so merging.

**Comment by ronaldtse** [comment]
Got it, thanks @ni4 !

**Pull request merged**